apps/src/applab/AppLabView.jsx

```diff
@@ -34,8 +35,101 @@ var AppLabView = React.createClass({
     onMount: React.PropTypes.func.isRequired
   },
 
+  getInitialState() {
+    // only used so that we can rerender when resized
+    return {
+      width: undefined,
+      height: undefined
+    };
+  },
+
+  /**
+   * Called when the window resizes. Look to see if width/height changed, then
+   * call adjustHeights as our maxHeight may need adjusting.
+   */
+  onResize: function () {
+    var windowWidth = $(window).width();
+    var windowHeight = $(window).height();
+
+    // We fire window resize events when the grippy is dragged so that non-React
+    // controlled components are able to rerender the editor. If width/height
+    // didn't change, we don't need to do anything else here
+    if (windowWidth === this.state.width && windowHeight === this.state.height) {
+      return;
+    }
+
+    this.setState({
+      width: $(window).width(),
+      height: $(window).height()
+    });
+
+    this.adjustHeights();
+  },
+
+  /**
+   * Adjust the height/maxHeight of our instructions based on the rendered size
+   * of the instructions, and the rendered size of the workspace.
+   */
+  adjustHeights: function () {
+    if (!this.props.instructionsInTopPane) {
+      return;
+    }
+
+    var instructionsContent = this.refs.topInstructions.refs.instructions.refs.instructionsMarkdown;
+    var instructionsContentHeight = $(ReactDOM.findDOMNode(instructionsContent)).outerHeight(true);
```

Comment:
Instead of reaching several layers down into child components, how about writing a `getContentHeight()` public method on `TopInstructions`.  You can invoke it from the ref:
```javascript
var instructionsContentHeight = this.refs.topInstructions.getContentHeight();
```